automation/tdx_canonical_setup.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+CUR_DIR=$(pwd)
+TDX_DIR="$CUR_DIR"/tdx_setup
+GUEST_TOOLS_DIR="$TDX_DIR"/guest-tools
+GUEST_IMG_DIR="$GUEST_TOOLS_DIR"/image
+ATTESTATION_DIR="$TDX_DIR"/attestation
+BRANCH_NAME=noble-24.04
+
+clone_tdx_repo(){
+        echo -e "\n\nCloning TDX git repository with '$BRANCH_NAME' branch.."
+        [ -d $TDX_DIR ] && rm -rf $TDX_DIR
+        git clone -b $BRANCH_NAME https://github.com/canonical/tdx.git $TDX_DIR
+}
+
+check_production_system() {
+        cd $ATTESTATION_DIR
+        echo -e "\n\nVerifying if the system supports attestation.."
+        output=$(sudo ./check-production.sh)
+        echo $output
+        if [[ $output =~ "Production" ]]; then
+                echo -e "Attestation is supported.\nThe attestation components will be installed "
+                sed -i 's/^TDX_SETUP_ATTESTATION=0/TDX_SETUP_ATTESTATION=1/' "$TDX_DIR"/setup-tdx-config
+        else
+                echo -e "Failure: Pre-production system. Attestation is not supported on Pre-production systems.\n"

```

Comment:
Not sure we should say failure as it should be okay if attestation is not supported on the system, some customers might just want to try TDX.

How about you say "This is a Pre-production system and hence Intel® SGX Data Center Attestation Primitives (Intel® SGX DCAP) will not be installed on the Host.

Reply:
updated the echo statement with new commit